smart-contracts/contracts/cl-vault/src/vault/swap.rs

```diff
@@ -1,21 +1,31 @@
-use std::str::FromStr;
-
 use apollo_cw_asset::AssetInfo;
 use cosmwasm_std::{
-    to_json_binary, Addr, Coin, CosmosMsg, DepsMut, Env, QuerierWrapper, Storage, Uint128, WasmMsg,
+    to_json_binary, Addr, Coin, CosmosMsg, DepsMut, Env, MessageInfo, QuerierWrapper, Response,
+    Storage, Uint128, WasmMsg,
 };
+use cw_dex_router::msg::{ExecuteMsg as ApolloExecuteMsg, QueryMsg as ApolloQueryMsg};
 use cw_dex_router::{
     msg::{BestPathForPairResponse, ExecuteMsg, QueryMsg},
     operations::SwapOperationsListUnchecked,
 };
 use osmosis_std::types::{
     cosmos::base::v1beta1::Coin as OsmoCoin, osmosis::poolmanager::v1beta1::SwapAmountInRoute,
 };
+use std::str::FromStr;
 
-use crate::{
-    state::{DEX_ROUTER, POOL_CONFIG},
-    ContractError,
-};
+use crate::state::DEX_ROUTER;
+use crate::{msg::SwapAsset, state::POOL_CONFIG};
+use crate::{state::VAULT_CONFIG, ContractError};
+
+use super::range::assert_range_admin;
+
+pub struct SwapCalculationResult {
```

Comment:
should swap amount here be token_in_amount?

Reply:
Yes, renaming